src/events/SDL_events.c

```diff
@@ -845,6 +943,7 @@ SDL_PushEvent(SDL_Event * event)
         return -1;
     }
 
+    SDL_SendWakeupEvent();
```

Comment:
I think this needs to be integrated deeper into the event processing code (maybe in `SDL_AddEvent()`). If someone calls `SDL_PeepEvents()` directly, it will not wake up a waiting thread.

Reply:
Actually for me this was a conscious decision, I thought the SDL_AddEvent and SDL_PeepEvents are sort-of low-level and they don't warrant a SendWakeupEvent. On the other side I see your point. If you think is better I can do this change.